hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -816,7 +830,10 @@ def text(
         )
     if (max_size == 0 or char_strategy.is_empty) and not min_size:
         return just("")
-    return TextStrategy(char_strategy, min_size=min_size, max_size=max_size)
+    # mypy is unhappy with ListStrategy(SearchStrategy[list[Ex]]) and then TextStrategy
+    # setting Ex = str. I think mypy is correct to think that str <: list[str]
+    # is incorrect and we probably do have an LSP violation here.
+    return TextStrategy(char_strategy, min_size=min_size, max_size=max_size)  # type: ignore
```

Comment:
Yeah, that's correct - we're using subclassing more for code-sharing than anything principled, in this particular case.